prow/plugins/size/size.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package size contains a Prow plugin which counts the number of lines changed
+// in a pull request, buckets this number into a few size classes (S, L, XL, etc),
+// and finally labels the pull request with this size.
+package size
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "size"
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pe github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.Logger, pe)
+}
+
+// Strict subset of *github.Client methods.
+type githubClient interface {
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetFile(org, repo, filepath, commit string) ([]byte, error)
+	GetPullRequestChanges(org, repo string, number int) ([]github.PullRequestChange, error)
+}
+
+func handlePR(gc githubClient, le *logrus.Entry, pe github.PullRequestEvent) error {
+	// These are the only two actions indicating the code diffs may have changed.
+	if pe.Action != "opened" && pe.Action != "synchronize" {
+		return nil
+	}
+
+	var (
+		owner = pe.PullRequest.Base.Repo.Owner.Login
+		repo  = pe.PullRequest.Base.Repo.Name
+		num   = pe.PullRequest.Number
+		sha   = pe.PullRequest.Base.SHA
+	)
+
+	g, err := NewGenFilesGroup(gc, owner, repo, sha)
+	if err != nil {
+		switch err.(type) {
+		case *ParseError:
+			// Continue on parse errors, but warn that something is wrong.
+			le.Warnf("error while parsing .generated_files: %v", err)
+		default:
+			return err
+		}
+	}
+
+	changes, err := gc.GetPullRequestChanges(owner, repo, num)
+	if err != nil {
+		return fmt.Errorf("GetPullRequestChanges error: %v", err)
+	}
+
+	var count int
+	for _, change := range changes {
+		if g.Match(change.Filename) {
+			continue
+		}
+
+		count += change.Additions + change.Deletions
+	}
+
+	labels, err := gc.GetIssueLabels(owner, repo, num)
+	if err != nil {
+		le.Warnf("while retrieving labels, error: %v", err)
```

Comment:
It seems that if you error out here, you don't get to run any of the following code so you might as well return here.